### PR TITLE
gh-107732: Mention dir support in importlib.resources docs

### PR DESCRIPTION
**Pull request by smheidrich**
Suggestion for how to point out `importlib.resources.as_file()`'s support for whole directories as resources in its docs.

* Issue: gh-107732

----
:books: Documentation preview :books:: https://cpython-previews--107734.org.readthedocs.build/

### CODE REVIEW

**Comment by smheidrich** [comment]
Ping @jaraco (as requested [here](https://discuss.python.org/t/importlib-resources-access-whole-directories-as-resources/15618/5?u=smheidrich))

**Review by jaraco — APPROVED** [review]

File: Doc/library/importlib.resources.rst [hunk 1]
```diff
@@ -82,15 +82,18 @@ for example, a package and its resources can be imported from a zip file using
...
 
     Use ``as_file`` when the Traversable methods
-    (``read_text``, etc) are insufficient and an actual file on
+    (``read_text``, etc) are insufficient and an actual file or directory on
     the file system is required.
 
     .. versionadded:: 3.9
+
+    .. versionchanged:: 3.12
+       Support added for ``traversable`` representing to a directory.
```
Comment (AA-Turner) [inline Doc/library/importlib.resources.rst]:
```suggestion
       Added support for ``traversable`` representing a directory.
```

**Review by AA-Turner — APPROVED** [review]

**Comment by smheidrich** [comment]
By the way, wouldn't it be appropriate to also mention this new capability of `as_file()` in the [Improved Modules section of 3.12's What's New page](https://docs.python.org/3.12/whatsnew/3.12.html#improved-modules)? Right now it only has an entry in the [full changelog](https://docs.python.org/3.12/whatsnew/changelog.html#id89), but there are some entries in Improved Modules that seem less important to mention than this to me (e.g. performance improvements). How are decisions on what does and doesn't go in there made? :thinking:

**Comment by AA-Turner** [comment]
Yep, we should add an entry to What's New. Are you happy to add it to this PR?

A

**Comment by smheidrich** [comment]
Sure :+1:  I added it.

File: Doc/whatsnew/3.12.rst [hunk 2]
```diff
@@ -611,6 +611,12 @@ fractions
 * Objects of type :class:`fractions.Fraction` now support float-style
   formatting. (Contributed by Mark Dickinson in :gh:`100161`.)
 
+importlib
+---------
```
Comment (AA-Turner) [inline Doc/whatsnew/3.12.rst]:
```suggestion
importlib.resources
-------------------
```

**Review by AA-Turner — APPROVED** [review]

**Review by hugovk — APPROVED** [review]

**Pull request merged**